CppCoreGuidelines.md

```diff
@@ -20358,7 +20358,7 @@ Individual classes are rarely a good logical unit of maintenance and distributio
 
 ##### Reason
 
-Splitting initialization into two leads to weaker invariants,
+Splitting initialization into two leads to less invariants,
```

Comment:
No, this is incorrect English. Fewer would be ok, but weak is still better IMHO.

Reply:
I would argue that in a general context weaker and fewer are not synonymous (taking fewer medications is not the same as taking weaker medications). In this example if anything it seems you have more invariants (in that you have invariants covering both phases of initialization) but the total effect of all invariant is reduced (i.e. they're weaker overall because the individual invariants are weaker).

As such I agree that "weaker" is the best option out of "less", "fewer" and "weaker".

I don't perceive the problem with "weaker" in this context. Here we don't use the term weak to mean bad. We use it to mean lacking in force. Certainly there is an implication that for an invariant to be lacking in force is a bad thing (because of the implications it as on code quality) but that seems fundamentally different to me than using the term "weak" synonymously with "bad".

Reply:
reverted